patches/chromium/feat_add_data_transfer_to_requestsingleinstancelock.patch

```diff
@@ -423,18 +423,30 @@ index ec725b44296266bea1a51aea889463a0bba8449c..a3d4dd1efc950033855a1f2783f94138
  
 -  *result = notification_callback.Run(parsed_command_line, current_directory) ?
 -      TRUE : FALSE;
++  // notification_callback.Run waits for StoreAck to
++  // run to completion before moving onwards.
++  // Therefore, we cannot directly send the SendBackAck
++  // callback instead, as it would hang the program
++  // during the ConnectNamedPipe call.
 +  g_write_ack_callback_called = false;
 +  *result = notification_callback.Run(parsed_command_line, current_directory,
 +                                      std::move(additional_data),
 +                                      base::BindRepeating(&StoreAck))
 +                ? TRUE
 +                : FALSE;
-+  g_ack_timer.Start(FROM_HERE, base::Seconds(0),
-+                    base::BindOnce(&SendBackAck));
++  if (*result) {
++    // If *result is TRUE, we return NOTIFY_SUCCESS.
++    // Only for that case does the second process read
++    // the acknowledgement. Therefore, only send back
++    // the acknowledgement if *result is TRUE,
++    // otherwise the program hangs during the ConnectNamedPipe call.
++    g_ack_timer.Start(FROM_HERE, base::Seconds(0),
```

Comment:
I am still not convinced for the purpose of the timer here, consider the following sequence,

* `ProcessLaunchNotification` gets called as soon as the second instance calls `chrome::AttemptToNotifyRunningChrome`
* `chrome::AttemptToNotifyRunningChrome` triggers `ReadAck` as soon as the notification to the message window is successful
* On the first instance side, `notification_callback.Run` represents [NotificationCallbackWrapper](https://github.com/electron/electron/blob/3d4d39d67b072a7c355dada2881cb6791e9e9f53/shell/browser/api/electron_api_app.cc#L518) which returns false only when the first instance is shutting down.

Problem:

1) `StoreAck` corresponds to [ack_callback](https://github.com/electron/electron/blob/3d4d39d67b072a7c355dada2881cb6791e9e9f53/shell/browser/api/electron_api_app.cc#L1121) which may get called by user once `preventDefault` action is executed if not we call it with `nullptr` for backwards compatibility. In the case, were user executes the `StoreAck` how does `SendBackAck` ensure the user input is obtained before piping data to the second instance. The timer fires as soon as `notification_callback.Run` is executed which does not correspond to user executing `StoreAck`.

2) `ReadAck` is called eagerly for all success scenarios irrespective of whether the first instance has data to send or not.

Solutions:

For supporting the cases of sending data from first instance to second instance, we should have a message window on the second instance as well and the first instance should notify this message window to prepare for reading data back once the first instance knows for sure there is data to send. This would eliminate the timer used here and also ensure `SendBackAck` is responsible for controlling the ack pipe creation.